include/measurement_kit/common/var.hpp

```diff
@@ -9,38 +9,61 @@
 
 namespace mk {
 
-template <typename T> class Var : public std::shared_ptr<T> {
-    using std::shared_ptr<T>::shared_ptr;
-
+template <typename T> class Var {
   public:
+    template <typename Deleter>
```

Comment:
If I get the jist of this correctly, you are now going for a template based/class composition approach vs class inheritance.
I believe this has as benefit the avoidance of object slicing. Is this correct?